scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -155,14 +155,38 @@ def test_rand_gen(self):
         # obtain a np.random.Generator object
         rng = np.random.default_rng(1)
 
-        res1 = dual_annealing(self.func, self.ld_bounds, seed=rng)
+        res1 = dual_annealing(self.func, self.ld_bounds, rng=rng)
         # seed again
         rng = np.random.default_rng(1)
-        res2 = dual_annealing(self.func, self.ld_bounds, seed=rng)
+        res2 = dual_annealing(self.func, self.ld_bounds, rng=rng)
         # If we have reproducible results, x components found has to
         # be exactly the same, which is not the case with no seeding
         assert_equal(res1.x, res2.x)
 
+    def test_rng_seed_spec007(self):
```

Comment:
We had discussed adding a generic tests for all of these. 

Perhaps just move this and the `differential_evolution` tests to `_lib` (ideally `match`ing the message associated with the error) and call it done?  If it was tested during the SPEC 7 process and it works for two SciPy functions, it will probably work for all of them.

The one thing you might want to keep for each function we modify is one or a few uses of the old keyword. When the decorator is removed, these will fail, and they can easily be replaced at that time.

Reply:
Duplication doesn't make sense. The only bits being removed at L174-->L188 though.

Reply:
I'm sorry, but I don't understand. If by "duplication" you mean having separate `test_rng_seed_spec007` for every function we decorate, I agree that duplication doesn't make sense, and I think this should be removed.

Reply:
I realised my comment wasn't clear.

Reply:
Agree I would prefer not to add a test to every function we do. Is leaving one use of `seed` going to cause a extra work when we turn on the deprecation though?

Reply:
Suppose that we don't think this through carefully and we realize later that if we had only added `test_rng_seed_spec007` rather than leaving an existing use of `seed` alone, we could have saved ourselves some trouble.

Then ISTM that in the worst case, we can change the use of `seed` to `rng` and add a `test_rng_seed_spec007` at that point. So in the worst case, we have time-shifted some work from now until later. What would be wrong with that?

But here is my attempt to think it through carefully now:

---
Assuming we require that each function has at least one test with its old keyword argument while the decorator is applied, either:
1) We add a test like this `test_rng_seed_spec007` (as it is now) or
2) We leave an existing use of the old keyword argument alone (as I'm proposing).

When we turn on the deprecations, we will have to add `with pytest.warns` context to something, whether it's a test like `test_rng_seed_spec007` or the existing use of the old keyword argument.

When we remove the decorator, we will have to change something, either removing a test like `test_rng_seed_spec007` or changing use of the old keyword argument to use of the new keyword argument.

However, if we go with option 2), we will have done less work overall because we never add a new `test_rng_seed_spec007` that only gets removed later.

Reply:
Spelled out like that option 2 is compelling to me